pkg/porter/cnab_test.go

```diff
@@ -355,3 +358,51 @@ func Test_bundleFileOptions(t *testing.T) {
 		})
 	}
 }
+
+func TestSharedOptions_populateInternalParameterSet(t *testing.T) {
+	p := NewTestPorter(t)
+	defer p.Teardown()
+
+	p.TestConfig.TestContext.AddTestFile("testdata/porter.yaml", config.Name)
+	m, err := manifest.LoadManifestFrom(context.Background(), p.Config, config.Name)
+	require.NoError(t, err)
+	bun, err := configadapter.ConvertToTestBundle(p.Context, m)
+	require.NoError(t, err)
+
+	sensitiveParamValue := "2"
+	nonsensitiveParamValue := "1"
+	opts := sharedOptions{}
+	opts.Params = []string{"my-first-param=" + nonsensitiveParamValue, "my-second-param=" + sensitiveParamValue}
+
+	err = opts.LoadParameters(p.Porter, bun)
+	require.NoError(t, err)
+
+	i := claims.NewInstallation("", bun.Name)
+
+	err = opts.populateInternalParameterSet(p.Porter, bun, &i)
+	require.NoError(t, err)
+
+	require.Len(t, i.Parameters.Parameters, 2)
+
+	// there should be no sensitive value on installation record
+	for _, param := range i.Parameters.Parameters {
+		if bun.IsSensitiveParameter(param.Name) {
```

Comment:
It makes perfect sense! I changed it to check for the correct data instead of relying on `IsSensitiveParameter`. I didn't make it into a table test since there isn't multiple test cases. Let me know if it works